File: algorithms/strings/urlify/urlify.ts
function urlify(str: string, true_len: number): string {
    let actualIdx = str.length;
    const arr = str.split('');

    for (let i = true_len - 1; i >= 0; i--) {
        if (arr[i] == ' ') {
            arr[actualIdx - 1] = '0';
            arr[actualIdx - 2] = '2';
            arr[actualIdx - 3] = '%';
            actualIdx -= 3;
        } else {
            arr[--actualIdx] = arr[i];
        }
    }

    return arr.join('');
}

const str = "Mr John Smith    ";
const encoded_str = urlify(str, 13);
console.log(`Encoded string : ${encoded_str}`);

File: algorithms/strings/urlify/urlify.rs
// TODO Try to do in-place

fn urlify(s: &str, _true_len: u8) -> String {
    let encoded_str = s.trim().replace(" ", "%20");
    return encoded_str;
}

fn main() {
    let s = String::from("Mr John Smith    ");

    let encoded_str = urlify(&s, 13);
    println!("Encoded string : {}", encoded_str);
}